test-suite/shared/src/test/require-jdk8/org/scalajs/testsuite/javalib/util/SplittableRandom.scala

```diff
@@ -0,0 +1,130 @@
+    assertEquals(-495782737, sr2.nextInt())
+    assertEquals(-1487672352, sr2.nextInt())
+    assertEquals(-538628223, sr2.nextInt())
+    assertEquals(1117712970, sr2.nextInt())
+    assertEquals(2081437683, sr2.nextInt())
+    assertEquals(2134440938, sr2.nextInt())
+    assertEquals(-2102672277, sr2.nextInt())
+    assertEquals(832521577, sr2.nextInt())
+    assertEquals(518494223, sr2.nextInt())
+    assertEquals(-42114979, sr2.nextInt())
+  }
+
+  @Test def should_correctly_implement_nextDouble(): Unit = {
+    val sr1 = new SplittableRandom(-45)
+    assertEquals(0.8229662358649753, sr1.nextDouble(), 0.0)
+    assertEquals(0.43324117570991283, sr1.nextDouble(), 0.0)
+    assertEquals(0.2639712712295723, sr1.nextDouble(), 0.0)
+    assertEquals(0.5576376282289696, sr1.nextDouble(), 0.0)
+    assertEquals(0.5505810186639037, sr1.nextDouble(), 0.0)
+    assertEquals(0.3944509738261206, sr1.nextDouble(), 0.0)
+    assertEquals(0.3108138671457821, sr1.nextDouble(), 0.0)
+    assertEquals(0.585951421265481, sr1.nextDouble(), 0.0)
+    assertEquals(0.2009547438834305, sr1.nextDouble(), 0.0)
+    assertEquals(0.8317691736686829, sr1.nextDouble(), 0.0)
+
+    val sr2 = new SplittableRandom(45)
+    assertEquals(0.9684135896502549, sr2.nextDouble(), 0.0)
+    assertEquals(0.9819686323309464, sr2.nextDouble(), 0.0)
+    assertEquals(0.5311927268453047, sr2.nextDouble(), 0.0)
+    assertEquals(0.8521356026917833, sr2.nextDouble(), 0.0)
+    assertEquals(0.01880601374789126, sr2.nextDouble(), 0.0)
+    assertEquals(0.37792881248018584, sr2.nextDouble(), 0.0)
+    assertEquals(0.7179744490511354, sr2.nextDouble(), 0.0)
+    assertEquals(0.3448879713662756, sr2.nextDouble(), 0.0)
+    assertEquals(0.023020123407108684, sr2.nextDouble(), 0.0)
+    assertEquals(0.6454709437764473, sr2.nextDouble(), 0.0)
+  }
+
+  @Test def should_correctly_implement_nextBoolean(): Unit = {
+    val sr1 = new SplittableRandom(4782934)
+    assertFalse(sr1.nextBoolean())
+    assertFalse(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+    assertFalse(sr1.nextBoolean())
+    assertFalse(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+    assertTrue(sr1.nextBoolean())
+
+    val sr2 = new SplittableRandom(-4782934)
+    assertFalse(sr2.nextBoolean())
+    assertFalse(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+    assertFalse(sr2.nextBoolean())
+    assertFalse(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+    assertTrue(sr2.nextBoolean())
+  }
+
+  @Test def should_correctly_implement_split(): Unit = {
+    val sr1 = new SplittableRandom(205620432625028L).split()
+    assertEquals(-2051870635339219700L, sr1.nextLong())
+    assertEquals(-1989219516, sr1.nextInt())
+    assertEquals(0.5689791577483271, sr1.nextDouble(), 0.0)
+  }
```

Comment:
`split()` seriously needs more testing. You only test 1 `split()` for 1 `seed`, and you do not test that the original instance of `SplittableRandom` (the receiver of `split()`) keeps working as expected after the split.

I'd like to see:
* Several input seeds.
* Testing the original `SplittableRandom` as much as the splitted one.
* One case with 2 levels of splits, where the 4 `SplittableRandom`s that are the result of 3 `split()`s in a "binary tree" keep working as expected.

Also, to test that they "work as expected" after the split, a better way is to test that 2 consecutive `nextLong()`s return the expected results. Indeed, as the paper explains on page 465, second column, the results of 2 consecutive `nextLong()`s provide complete knowledge about the internal state of the `SplittableRandom`. So if 2 consecutive `nextLong()`s succeed after the split, we know that `split()` correctly updated the internal state. Testing `nextInt()` and `nextDouble()` is not useful here.